scripts/tools/metricTracer.yml

```diff
@@ -0,0 +1,255 @@
+type: function
+desc: "Create a metric tracer for a device."
+version: "1.10"
+class: $tMetricTracer
+name: CreateExp
+decl: static
+details:
+    - "The notification event must have been created from an event pool that was created using $X_EVENT_POOL_FLAG_HOST_VISIBLE flag."
+    - "The duration of the signal event created from an event pool that was created using $X_EVENT_POOL_FLAG_KERNEL_TIMESTAMP flag is undefined. However, for consistency and orthogonality the event will report correctly as signaled when used by other event API functionality."
+    - "The application must **not** call this function from simultaneous threads with the same device handle."
+    - "The metric tracer is created in disabled state"
+    - "Metric groups must be of the sampling type ZET_METRIC_SAMPLING_TYPE_EXP_FLAG_TRACER_BASED"
+    - "All metric groups must be first activated"
+params:
+    - type: $t_context_handle_t
+      name: hContext
+      desc: "[in] handle of the context object"
+    - type: "$t_device_handle_t"
+      name: hDevice
+      desc: "[in] handle of the device"
+    - type: uint32_t
+      name: "metricGroupCount"
+      desc: "[in] metric group count"
+    - type: "$t_metric_group_handle_t"
+      name: "hMetricGroups"
+      desc: "[in][range(0, metricGroupCount )] handles of the metric groups to trace"
+    - type: "$t_metric_tracer_exp_desc_t*"
+      name: desc
+      desc: "[in,out] metric tracer descriptor"
+    - type: "$x_event_handle_t"
+      name: hNotificationEvent
+      desc: "[in][optional] event used for report availability notification. Note: If buffer is not drained when the event it flagged, there is a risk of HW event buffer being overrun"
+    - type: "$t_metric_tracer_exp_handle_t*"
+      name: phMetricTracer
+      desc: "[out] handle of the metric tracer"
+returns:
+    - $X_RESULT_ERROR_INVALID_SYNCHRONIZATION_OBJECT
+--- #--------------------------------------------------------------------------
+type: function
+desc: "Destroy a metric tracer."
+version: "1.10"
+class: $tMetricTracer
+name: DestroyExp
+details:
+    - "The application must **not** call this function from simultaneous threads with the same metric tracer handle."
+params:
+    - type: "$t_metric_tracer_exp_handle_t"
+      name: hMetricTracer
+      desc: "[in] handle of the metric tracer"
+--- #--------------------------------------------------------------------------
+type: function
+desc: "Lightweight call that starts the event collections"
+version: "1.10"
+class: $tMetricTracer
+name: EnableExp
+details:
+    - "The application must **not** call this function from simultaneous threads with the same metric tracer handle."
+params:
+    - type: "$t_metric_tracer_exp_handle_t*"
+      name: phMetricTracer
+      desc: "[in] handle of the metric tracer"
+    - type: $x_bool_t
+      name: synchronous
+      desc: "[in] request synchronous behavior. Confirmation of successful asynchronous operation is done by calling $tMetricTracerReadDataExp()"
```

Comment:
Please clarify if ReadData with 0 data size enable/disable the tracer ?

Reply:
Note that the confirmation is based on the return call from the API, not the size. When going from inactive to active, will return NOT_READY (obviously 0 size data) until SUCCESS and may or may not have data.  
When going from active to inactive will return SUCESS  while active until get NOT READY.